sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/json/rapids/GpuJsonScan.scala

```diff
@@ -103,12 +104,24 @@ object GpuJsonScan {
   }
 
   def tagJsonToStructsSupport(options:Map[String, String],
-                              meta: RapidsMeta[_, _, _]): Unit = {
+      dt: DataType,
+      meta: RapidsMeta[_, _, _]): Unit = {
     val parsedOptions = new JSONOptionsInRead(
       options,
       SQLConf.get.sessionLocalTimeZone,
       SQLConf.get.columnNameOfCorruptRecord)
 
+    meta.tagForGpu()
+
+    val hasDates = TrampolineUtil.dataTypeExistsRecursively(dt, _.isInstanceOf[DateType])
+    if (hasDates) {
+      options.get("dateFormat") match {
```

Comment:
nit: I think this is fine. It appears to match the Spark code exactly. But it might be nice to use an actual JSONOptions, even if we have to go through a trampoline, and get dateFormatInRead to know for sure what Spark thinks it is.